app/src/main/java/co/smartreceipts/android/widget/tooltip/report/ReportTooltipFragment.java

```
@@ -0,0 +1,130 @@
+            presentGenerateInfo();
+        } else if (uiIndicator.getState() == None) {
+            tooltip.hideWithAnimation();
+        }
+    }
+
+    @NonNull
+    @Override
+    public Observable<ReportTooltipUiIndicator> getCloseButtonClicks() {
+        return closeClickStream;
+    }
+
+    @NonNull
+    @Override
+    public Observable<ReportTooltipUiIndicator> getTooltipsClicks() {
+        return tooltipClickStream;
+    }
+
+    private void presentError(@NonNull final SyncErrorType syncErrorType) {
+        if (syncErrorType == SyncErrorType.UserRevokedRemoteRights) {
+            tooltip.setErrorWithoutClose(R.string.drive_sync_error_no_permissions, view ->
+                    tooltipClickStream.onNext(ReportTooltipUiIndicator.syncError(syncErrorType)));
+        } else if (syncErrorType == SyncErrorType.UserDeletedRemoteData) {
+            tooltip.setErrorWithoutClose(R.string.drive_sync_error_lost_data, view -> {
+                tooltipClickStream.onNext(ReportTooltipUiIndicator.syncError(syncErrorType));
+            });
+        } else if (syncErrorType == SyncErrorType.NoRemoteDiskSpace) {
+            tooltip.setError(R.string.drive_sync_error_no_space, view -> {
+                closeClickStream.onNext(ReportTooltipUiIndicator.syncError(syncErrorType));
+            });
+        }
+        tooltip.showWithAnimation();
+    }
+
+    private void presentGenerateInfo() {
+        tooltip.setInfo(R.string.tooltip_generate_info_message,
+                v -> {
+                    tooltipClickStream.onNext(ReportTooltipUiIndicator.generateInfo());
+
+                    Fragment parentFragment = this.getParentFragment();
+                    if (parentFragment instanceof GenerateNavigator) {
+                        ((GenerateNavigator) parentFragment).navigateToGenerateTab();
+                    } else {
+                        throw new IllegalStateException("Parent fragment must implement GenerateNavigator interface");
```

Let's move this check into `onCreate` so we fail fast here